kitsune/products/views.py

```diff
@@ -45,18 +57,39 @@ def product_landing(request, slug):
             latest_version = versions[0].min_version
         else:
             latest_version = 0
+    topics = topics_for(request.user, product=product, parent=None)
+    # Create a dictionary of topics and three documents
+    # TODO: get popular documents instead of just the first three
+    documents = {}
+    topics_with_urls = []
+    for topic in topics:
+        docs, _ = documents_for(
+            request.user, request.LANGUAGE_CODE, topics=[topic], products=[product]
+        )
+        total_articles = len(docs)
+        # Create a dict with topic data including the URL
+        topic_data = {
+            "topic": topic,
+            "topic_url": reverse("products.documents", args=[product.slug, topic.slug]),
+            "title": topic.title,
+            "total_articles": total_articles,
+            "image_url": getattr(topic, "image_url", ""),
+        }
+        topics_with_urls.append(topic_data)
+        documents[topic] = docs[:3]
```

Comment:
It seems that you could toss the list of documents into the `topic_data` dict, and then you wouldn't have to pass the `documents` dict into the template context nor the `help_topics()` macro.

Reply:
Thanks - totally obvious now.